build.gradle

```diff
@@ -0,0 +1,681 @@
+    compile group: 'commons-lang', name: 'commons-lang', version: '2.6'
+    compile group: 'net.sf.json-lib', name: 'json-lib', version: '2.4', classifier: 'jdk15'
+    compile group: 'org.apache.lucene', name: 'lucene-core', version: '2.9.4'
+    compile group: 'com.itextpdf', name: 'itextpdf', version: '5.5.9'
+    compile group: 'javax.ejb', name: 'javax.ejb-api', version: '3.2'
+    compile group: 'org.freemarker', name: 'freemarker', version: '2.3.20'
+}
+
+def buildAliases = [
+        'gnomex_all' : ['clean', 'clean_tomcat', 'enableHTTPS', 'makeFlex', 'makeSQLServer', 'jar', 'clientJar', 'realmJar', 'flex-compile', 'copyScripts', 'war', 'gnomex_deploy'],
+        'gnomex_all_opensource' : ['clean', 'clean_tomcat', 'disableHTTPS', 'makeMySQL', 'makeFlex', 'jar', 'clientJar', 'realmJar', 'flex-compile', 'copyScripts', 'opensourceWar', 'enableHTTPS', 'opensourceSecureWar', 'moveSecureWar', 'gnomex_deploy', 'createOpenSourceZip'],
+        'gnomex_clean' : ['clean', 'clean_tomcat'],
+        'gnomex_lite_all' : ['clean', 'clean_tomcat', 'enableHTTPS', 'makeLite', 'makeSQLServer', 'jar', 'clientJar', 'realmJar', 'copyScripts', 'gnomexLiteWar', 'gnomex_deploy'],
+        'gnomex_lite_all_opensource' : ['clean', 'clean_tomcat', 'disableHTTPS', 'makeLite', 'makeMySQL', 'jar', 'clientJar', 'realmJar', 'copyScripts', 'gnomexLiteOpenSourceWar', 'enableHTTPS', 'gnomexLiteOpenSourceSecureWar', 'moveSecureWar', 'gnomex_deploy']
+]
+def expandedTaskList = []
+
+gradle.startParameter.taskNames.each {
+    expandedTaskList << (buildAliases[it] ? buildAliases[it] : it)
+}
+
+gradle.startParameter.taskNames = expandedTaskList.flatten()
+
+task clean_tomcat << {
+    delete "${serverHome}/webapps/gnomex.war"
+    delete "${serverHome}/lib/gnomex_realm.jar"
+    delete "${serverHome}/webapps/gnomex"
+    delete "${serverHome}/webapps/gnomexlite.war"
+    delete "${serverHome}/webapps/gnomexLite"
+}
+
+// replace string  helper function
+def replaceString(fileName, fromString, toString) {
+    def file = new File(fileName)
+    def fileText = file.text
+    fileText = fileText.replace(fromString, toString);
+    new File(fileName).write(fileText)
+}
+
+task moveSecureWar << {
+    copy{
+        from buildDir.getAbsolutePath() + "/libs"
+        include "*secure*"
+        into buildDir.getAbsolutePath() + "/secure";
+        rename("gnomexlite_secure.war", "gnomexlite.war");
+        rename("gnomex_secure.war", "gnomex.war");
+    }
+}
+
+task disableHTTPS << {
+    replaceString("${baseDir}/src/main/java/hci/gnomex/constants/Constants.java",
+            "REQUIRE_SECURE_REMOTE = true;",
+            "REQUIRE_SECURE_REMOTE = false;")
+}
+
+task enableHTTPS << {
+    replaceString("${baseDir}/src/main/java/hci/gnomex/constants/Constants.java",
+            "REQUIRE_SECURE_REMOTE = false;",
+            "REQUIRE_SECURE_REMOTE = true;")
+}
+
+task makeFlex <<{
+    replaceString("${webrootDir}/WEB-INF/web.xml", "gnomexLite.jsp", "gnomexFlex.jsp");
```

Comment:
Does this string replacement result in Git thinking the file is edited? There a technique that can be applied where all these files that are modified at build time can essentially be "templates" that contain placeholders for the modification, or you modify the files as the war is build so that you only change the in-flight file going into the build product and not the version controlled file.